### PR TITLE
chore(deps): swap out pretty_assertions for similar-asserts

### PR DESCRIPTION
**Pull request by lukesteensen**
The previous crate would take wayyy to long on large inputs, show diffs that were much too large, etc. New crate is much improved.

Signed-off-by: Luke Steensen <luke.steensen@gmail.com>

### CODE REVIEW

**Review by neuronull — APPROVED** [review]
LGTM

File: lib/vector-common/src/event_data_eq.rs [hunk 1]
```diff
@@ -18,7 +18,12 @@ macro_rules! assert_event_data_eq {
                     left.event_data_eq(right),
                     "assertion failed: {}\n\n{}\n",
                     $message,
-                    pretty_assertions::Comparison::new(left, right),
+                    similar_asserts::SimpleDiff::from_str(
```
Comment (spencergilbert) [inline lib/vector-common/src/event_data_eq.rs:21]: This is definitely uglier, is there a better way we could compare?

**Review by spencergilbert — APPROVED** [review]
Was there a comparison of the perf between the two crates somewhere, or just from us testing (locally?)?

File: lib/vector-common/src/event_data_eq.rs (hunk 1 above)
Comment (lukesteensen) [reply lib/vector-common/src/event_data_eq.rs:21]: This is actually what `pretty_assertions` was [doing already](https://github.com/rust-pretty-assertions/rust-pretty-assertions/blob/27ee62821f4c812e56484c9cc73ff7f27382a02a/pretty_assertions/src/lib.rs#L128-L141).

Comment (spencergilbert) [reply lib/vector-common/src/event_data_eq.rs:21]: Haha, but I don't have to see it! Obviously not a big thing.

**Comment by lukesteensen** [comment]
> Was there a comparison of the perf between the two crates somewhere, or just from us testing (locally?)?

Good question. Here's with this patch for a failing test I'm working on:

```
cargo test --lib event::test::serialization::back_and_forth_through_bytes  0.28s user 0.15s system 92% cpu 0.464 total
```
And without it, I gave up waiting:

```
running 1 test
test event::test::serialization::back_and_forth_through_bytes has been running for over 60 seconds
^C
cargo test --lib event::test::serialization::back_and_forth_through_bytes  6.29s user 2.58s system 0% cpu 34:35.51 total
```

So it's pretty dramatic for these massive inputs. There's also an equally big benefit in that in only emits the area around the differences:

```
Differences (-left|+right):
                                     n: 45473,
                                 },
                             ],
                             count: 0,
-                            min: 211136.0,
-                            max: 662144.0,
-                            sum: -356288.0,
-                            avg: -128896.0,
+                            min: 1.7976931348623157e308,
+                            max: -1.7976931348623157e308,
+                            sum: 0.0,
+                            avg: 0.0,
                         },
                     ),
                 },
             },
```

This is way more helpful than includes the entire (massive) structure, which actually causes Github Actions to choke on the output.

**Review by bruceg — APPROVED** [review]

**Pull request merged**